include/boost/math/tools/promotion.hpp

```diff
@@ -78,14 +96,35 @@ namespace boost
 #ifdef BOOST_MATH_USE_FLOAT128
            typename std::conditional<std::is_same<__float128, T1P>::value || std::is_same<__float128, T2P>::value, // either long double?
             __float128,
-#endif
+#endif 
+#ifdef __STDCPP_FLOAT128_T__
+           typename std::conditional<std::is_same<std::float128_t, T1P>::value || std::is_same<std::float128_t, T2P>::value, // either long double?
```

Comment:
I wonder if now that we have `std::common_type`, whether we should basically just rely on that here?

It simplifies the code, but also, it handles the case where double or long double do not have the sizes we expect.

Reply:
I could try it. Since `std::common_type` is for implicit conversion I wonder if it will work with these types since they require explicit conversions.